File: observerpattern/src/main/java/com/example/observerpattern/MainActivity.java
package com.example.observerpattern;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;


/*
 * 練習Observer Pattern
 * observer 觀察者，訂閱者
 * subject(Observable) 主體/目標 ，被訂閱者
 * 1.建立Observer和Observable 的 interface
 * 2.建立一個 new 去implement Observable
 * 3.建立一個 Reader 去implement Observer
 * 4.建立一個 Reader 和 New 變數
 * 5.Reader 變數去 訂閱 new的變數
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}

File: kotlin/src/main/java/com/example/kotlin/MainActivity.kt
package com.example.kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import java.time.LocalDate
import java.time.LocalTime
import java.time.format.DateTimeFormatter
import java.time.format.FormatStyle

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val textview = findViewById<TextView>(R.id.textview)
        var time = LocalTime.parse("2:11 PM", DateTimeFormatter.ofLocalizedTime(FormatStyle.SHORT))
        println("這裡啊>>>" + time)
    }
}

